src/test/java/de/tum/in/www1/artemis/metis/PostIntegrationTest.java

```diff
@@ -529,6 +531,11 @@ void testGetPostsForCourse() throws Exception {
         // get amount of posts with that certain
         database.assertSensitiveInformationHidden(returnedPosts);
         assertThat(returnedPosts).hasSameSizeAs(existingCoursePosts);
+
+        assertThat(returnedPosts.stream().filter(post -> Arrays.asList(1L, 2L, 3L, 4L).contains(post.getId())).allMatch(post -> post.getAuthorRole().equals(UserRole.USER)));
+        assertThat(returnedPosts.stream().filter(post -> Arrays.asList(5L, 6L, 7L, 8L, 10L, 11L, 12L, 13L).contains(post.getId()))
+                .allMatch(post -> post.getAuthorRole().equals(UserRole.TUTOR)));
+        assertThat(returnedPosts.stream().noneMatch(post -> post.getAuthorRole().equals(UserRole.INSTRUCTOR)));
```

Comment:
You currently don't assert anything here.
```suggestion
        assertThat(returnedPosts.stream().filter(post -> Arrays.asList(1L, 2L, 3L, 4L).contains(post.getId()))).allMatch(post -> post.getAuthorRole().equals(UserRole.USER));
        assertThat(returnedPosts.stream().filter(post -> Arrays.asList(5L, 6L, 7L, 8L, 10L, 11L, 12L, 13L).contains(post.getId())))
                .allMatch(post -> post.getAuthorRole().equals(UserRole.TUTOR));
        assertThat(returnedPosts).noneMatch(post -> post.getAuthorRole().equals(UserRole.INSTRUCTOR));
```

Reply:
sharp eyes! 👀